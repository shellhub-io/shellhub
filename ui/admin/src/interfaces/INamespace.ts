import { INamespace, INamespaceMember } from "@/interfaces/INamespace";

type AdminUserStatus = "accepted" | "pending";
interface IAdminBilling {
  active: boolean;
  current_period_end: string;
  customer_id: string;
  payment_failed: null | string | boolean;
  payment_method_id: string;
  price_id: string;
  state: string;
  sub_item_id: string;
  subscription_id: string;
}

interface IAdminNamespaceMember extends Omit<INamespaceMember, "status"> {
  status: AdminUserStatus
}

export interface IAdminNamespace extends Omit<INamespace, "billing" | "members"> {
  billing?: IAdminBilling;
  members: IAdminNamespaceMember[];
}
