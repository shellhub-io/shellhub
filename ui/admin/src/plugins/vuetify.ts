// Styles
// import "@mdi/font/css/materialdesignicons.css";
import "vuetify/styles";

// Vuetify
import { createVuetify } from "vuetify";

export default createVuetify({
  theme: {
    defaultTheme: "dark",
  },
});
