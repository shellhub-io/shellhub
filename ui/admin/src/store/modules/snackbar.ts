import { Module } from "vuex";
import { State } from "./../index";

export interface snackbarMessageAndContentType {
  typeMessage: string;
  typeContent: string;
}

export interface SnackbarState {
  snackbarError: boolean;
  snackbarSuccess: boolean;
  snackbarMessageAndContentType: snackbarMessageAndContentType;
  snackbarCopy: boolean;
}

export const snackbar: Module<SnackbarState, State> = {
  namespaced: true,

  state: {
    snackbarError: false,
    snackbarSuccess: false,
    snackbarMessageAndContentType: { typeMessage: "", typeContent: "" },
    snackbarCopy: false,
  },

  getters: {
    snackbarSuccess: (state) => state.snackbarSuccess,
    snackbarError: (state) => state.snackbarError,
    snackbarMessageAndContentType: (state) => state.snackbarMessageAndContentType,
    snackbarCopy: (state) => state.snackbarCopy,
  },

  mutations: {
    setSnackbarSuccessAction: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarSuccess = true;
    },

    setSnackbarNoContent: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarSuccess = true;
    },

    setSnackbarSuccessDefault: (state) => {
      state.snackbarMessageAndContentType = { typeMessage: "default", typeContent: "" };
      state.snackbarSuccess = true;
    },

    unsetSnackbarSuccess: (state) => {
      state.snackbarSuccess = false;
    },

    setSnackbarErrorLoadingOrAction: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarError = true;
    },

    setSnackbarErrorCustom: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarError = true;
    },

    setSnackbarErrorDefault: (state) => {
      state.snackbarMessageAndContentType = { typeMessage: "default", typeContent: "" };
      state.snackbarError = true;
    },

    setSnackbarErrorLicense: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarError = true;
    },

    unsetSnackbarError: (state) => {
      state.snackbarError = false;
    },

    setSnackbarCopy: (state, value) => {
      state.snackbarMessageAndContentType = { typeMessage: "", typeContent: value };
      state.snackbarCopy = true;
    },

    unsetSnackbarCopy: (state) => {
      state.snackbarCopy = false;
    },
  },

  actions: {
    showSnackbarSuccessAction: ({ commit }, value) => {
      const data = { typeMessage: "action", typeContent: value };
      commit("setSnackbarSuccessAction", data);
    },

    showSnackbarSuccessDefault: ({ commit }) => {
      commit("setSnackbarSuccessDefault");
    },

    unsetShowStatusSnackbarSuccess: ({ commit }) => {
      commit("unsetSnackbarSuccess");
    },

    showSnackbarErrorLoading: ({ commit }, value) => {
      const data = { typeMessage: "loading", typeContent: value };
      commit("setSnackbarErrorLoadingOrAction", data);
    },

    showSnackbarNoContent: ({ commit }) => {
      const data = { typeMessage: "no-content", typeContent: "" };
      commit("setSnackbarNoContent", data);
    },

    showSnackbarErrorAction: ({ commit }, value) => {
      const data = { typeMessage: "action", typeContent: value };
      commit("setSnackbarErrorLoadingOrAction", data);
    },

    showSnackbarErrorCustom: ({ commit }, value) => {
      const data = { typeMessage: "custom", typeContent: value };
      commit("setSnackbarErrorCustom", data);
    },

    showSnackbarErrorLicense: ({ commit }, value) => {
      const data = { typeMessage: "licenseRequired", typeContent: value };
      commit("setSnackbarErrorLicense", data);
    },

    showSnackbarErrorDefault: ({ commit }) => {
      commit("setSnackbarErrorDefault");
    },

    unsetShowStatusSnackbarError: ({ commit }) => {
      commit("unsetSnackbarError");
    },

    showSnackbarCopy: ({ commit }, value) => {
      commit("setSnackbarCopy", value);
    },

    unsetShowStatusSnackbarCopy: ({ commit }) => {
      commit("unsetSnackbarCopy");
    },
  },
};
