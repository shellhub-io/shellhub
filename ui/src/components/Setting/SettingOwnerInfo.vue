<template>
  <div class="text-center">
    <div data-test="message-div">
      <h3 class="pl-6">
        <span>You're not the owner of this namespace.</span>
        <p data-test="contactUser-p">Contact the owner for more information.</p>
      </h3>
    </div>
    <br />
  </div>
</template>

<script setup lang="ts">
</script>
