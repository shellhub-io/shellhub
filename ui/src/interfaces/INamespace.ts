export type BasicRole = "administrator" | "operator" | "observer";
export type Role = BasicRole | "owner";

export interface INamespaceMember {
  id: string;
  role: Role;
  email: string;
  status: string;
  added_at: string;
  expires_at: string;
}

export interface INamespaceSettings {
  connection_announcement?: string;
  session_record: boolean;
}

interface INamespaceBilling {
  customer_id: string;
  subscription_id: string;
  current_period_end: number;
  active: boolean;
  status: string;
}

export interface INamespace {
  name: string;
  owner: string;
  tenant_id: string;
  members: INamespaceMember[];
  settings: INamespaceSettings;
  max_devices: number;
  devices_accepted_count: number;
  devices_pending_count: number;
  devices_rejected_count: number;
  created_at: string;
  billing: INamespaceBilling | null;
}

export interface INamespaceAcceptInvite {
  tenant: string;
  sig: string;
}

export interface INamespaceAddMember {
  email: string;
  role: Role;
  tenant_id: string;
}

export interface INamespaceEditMember {
  user_id: string;
  role: Role;
  tenant_id: string;
}

export interface INamespaceRemoveMember {
  tenant_id: string;
  user_id: string;
}

export interface INamespaceEdit {
  tenant_id: string;
  name?: string;
  settings?: Partial<INamespaceSettings>;
}
