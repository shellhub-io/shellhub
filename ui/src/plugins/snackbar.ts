import { App, Plugin, reactive } from "vue";

const SnackbarInjectionKey = Symbol("snackbar");

type SnackbarType = "success" | "error" | "info" | "warning";

interface SnackbarState {
  message: string;
  type: SnackbarType;
  show: boolean;
}

interface ISnackbarPlugin {
  showSuccess: (msg: string) => void;
  showError: (msg: string) => void;
  showInfo: (msg: string) => void;
  showWarning: (msg: string) => void;
  getMessage: () => string;
  getType: () => SnackbarType;
  getShow: () => boolean;
}

const state = reactive<SnackbarState>({
  message: "",
  type: "info",
  show: false,
});

const showSnackbar = (type: SnackbarType, message: string) => {
  state.message = message;
  state.type = type;
  state.show = true;

  setTimeout(() => {
    state.show = false;
  }, 4000);
};

const plugin: ISnackbarPlugin = {
  showSuccess: (msg: string) => showSnackbar("success", msg),
  showError: (msg: string) => showSnackbar("error", msg),
  showInfo: (msg: string) => showSnackbar("info", msg),
  showWarning: (msg: string) => showSnackbar("warning", msg),
  getMessage: () => state.message,
  getType: () => state.type,
  getShow: () => state.show,
};

const SnackbarPlugin: Plugin = {
  install(app: App) {
    app.provide(SnackbarInjectionKey, plugin);
  },
};

export { SnackbarPlugin, SnackbarInjectionKey, plugin };
export type { SnackbarState, ISnackbarPlugin };
