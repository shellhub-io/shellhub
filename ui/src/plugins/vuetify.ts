// Styles
import "@mdi/font/css/materialdesignicons.css";
import "@fortawesome/fontawesome-free/css/all.css";
import "vuetify/styles";
import { VFileUpload, VFileUploadItem } from "vuetify/labs/VFileUpload";
import "../styles/_variables.scss";
import { aliases as faAliases, fa } from "vuetify/iconsets/fa";

// Vuetify
import { createVuetify } from "vuetify";

const light = {
  dark: false,
  colors: {
    primary: "#667acc",
    secondary: "#FFFFFF",
    background: "#F5F5F5",
    tabs: "#F5F5F5",
    foreground: "#F5F5F5",
    paymentForm: "#F5F5F5",
    terminal: "#0f1526",
    "v-theme-background": "#FFFFFF",
    "v-theme-surface": "#FFFFFF",
    "v-theme-card": "#FFFFFF",
  },
};

const dark = {
  dark: true,
  colors: {
    primary: "#667acc",
    secondary: "#1E2127",
    background: "#18191B",
    tabs: "#1E1E1E",
    foreground: "#1E1E1E",
    paymentForm: "#E0E0E0",
    terminal: "#0f1526",
    "v-theme-background": "#1E2127",
    "v-theme-surface": "#1E2127",
    "v-theme-card": "#22252B",
  },
};

export default createVuetify({
  components: {
    VFileUpload,
    VFileUploadItem,
  },
  theme: {
    defaultTheme: "dark",
    themes: {
      dark,
      light,
    },
  },
  defaults: {
    global: {

    },
    VTextField: {
      variant: "outlined",
      density: "comfortable",
    },
    VSelect: {
      variant: "outlined",
      density: "comfortable",
    },
    VTextarea: {
      variant: "outlined",
    },
  },
  icons: {
    faAliases,
    sets: {
      fa,
    },
  },
});
