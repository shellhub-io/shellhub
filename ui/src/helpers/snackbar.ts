import { inject } from "vue";
import { InjectionKey, type ISnackbarPlugin } from "@/plugins/snackbar";

const useSnackbar = (): ISnackbarPlugin => inject(InjectionKey) as ISnackbarPlugin;

export default useSnackbar;
