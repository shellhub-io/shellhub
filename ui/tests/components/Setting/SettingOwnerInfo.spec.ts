import { setActivePinia, createPinia } from "pinia";
import { createVuetify } from "vuetify";
import { mount, VueWrapper } from "@vue/test-utils";
import { beforeEach, describe, expect, it } from "vitest";
import SettingOwnerInfo from "@/components/Setting/SettingOwnerInfo.vue";
import { SnackbarPlugin } from "@/plugins/snackbar";
import useNamespacesStore from "@/store/modules/namespaces";
import { INamespaceMember } from "@/interfaces/INamespace";

type SettingOwnerInfoWrapper = VueWrapper<InstanceType<typeof SettingOwnerInfo>>;

const members = [
  {
    id: "507f1f77bcf86cd799439011",
    role: "owner" as const,
  },
  {
    id: "507f1f77bcf86cd799745632",
    role: "operator" as const,
  },
] as INamespaceMember[];

const namespaceData = {
  name: "test",
  owner: "507f1f77bcf86cd799439011",
  tenant_id: "fake-tenant-data",
  members,
  max_devices: 3,
  devices_count: 3,
  created_at: "",
  billing: null,
  settings: {
    session_record: true,
  },
  devices_accepted_count: 3,
  devices_rejected_count: 0,
  devices_pending_count: 0,
};

describe("Setting Owner Info", () => {
  let wrapper: SettingOwnerInfoWrapper;
  setActivePinia(createPinia());
  const namespacesStore = useNamespacesStore();
  const vuetify = createVuetify();

  beforeEach(async () => {
    namespacesStore.currentNamespace = namespaceData;

    wrapper = mount(SettingOwnerInfo, {
      global: {
        plugins: [vuetify, SnackbarPlugin],
      },
    });
  });

  it("Is a Vue instance", () => {
    expect(wrapper.vm).toBeTruthy();
  });

  it("Renders the component", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Displays message when user is not the owner", async () => {
    expect(wrapper.find('[data-test="message-div"]').exists()).toBe(true);
    expect(wrapper.find('[data-test="contactUser-p"]').text()).toContain("Contact the owner for more information.");
  });
});
