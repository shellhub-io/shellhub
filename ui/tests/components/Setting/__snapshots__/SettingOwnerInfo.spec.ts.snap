// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Setting Owner Info > Renders the component 1`] = `
"<div class="text-center">
  <div data-test="message-div">
    <h3 class="pl-6"><span>You're not the owner of this namespace.</span>
      <p data-test="contactUser-p">Contact the owner for more information.</p>
    </h3>
  </div><br>
</div>"
`;
